community/bolt/src/test/java/org/neo4j/bolt/v1/runtime/BoltStateMachineTest.java

```diff
@@ -401,4 +411,39 @@ public void actionsDisallowedBeforeInitialized() throws Throwable
         }
     }
 
+    @SuppressWarnings( "unchecked" )
+    @Test
+    public void shouldTerminateOnAuthExpiryDuringREADYRun() throws Throwable
+    {
+        // Given
+        TransactionStateMachine.SPI transactionSPI = mock( TransactionStateMachine.SPI.class );
+        when( transactionSPI.beginTransaction( any() ) ).thenThrow( new AuthExpirationException( "Auth expired!" ) );
```

Comment:
I kinda liked the below test call better, with `doThrow(...).when(...)`. Is there a reason it's asymmetrical like this here?